pkg/operation/botanist/controlplane/kubescheduler/monitoring.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubescheduler
+
+import (
+	"bytes"
+	"text/template"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+const (
+	monitoringPrometheusJobName = "kube-scheduler"
+
+	monitoringMetricSchedulerBindingLatencyMicrosecondsBucket             = "scheduler_binding_latency_microseconds_bucket"
+	monitoringMetricSchedulerE2eSchedulingLatencyMicrosecondsBucket       = "scheduler_e2e_scheduling_latency_microseconds_bucket"
+	monitoringMetricSchedulerSchedulingAlgorithmLatencyMicrosecondsBucket = "scheduler_scheduling_algorithm_latency_microseconds_bucket"
+	monitoringMetricRestClientRequestsTotal                               = "rest_client_requests_total"
+	monitoringMetricProcessMaxFds                                         = "process_max_fds"
+	monitoringMetricProcessOpenFds                                        = "process_open_fds"
+
+	// TODO: Replace below hard-coded paths to Prometheus certificates once its deployment has been refactored.
+	monitoringScrapeConfigTmpl = `- job_name: ` + monitoringPrometheusJobName + `
+  {{- if .k8sGreaterEqual113 }}
+  scheme: https
+  tls_config:
+    insecure_skip_verify: true
+    cert_file: /etc/prometheus/seed/prometheus.crt
+    key_file: /etc/prometheus/seed/prometheus.key
+  {{- end }}
+  honor_labels: false
+  kubernetes_sd_configs:
+  - role: endpoints
+    namespaces:
+      names: [{{ .namespace }}]
+  relabel_configs:
+  - source_labels:
+    - __meta_kubernetes_service_name
+    - __meta_kubernetes_endpoint_port_name
+    action: keep
+    regex: ` + ServiceName + `;` + PortNameMetrics + `
+  - action: labelmap
+    regex: __meta_kubernetes_service_label_(.+)
+  - source_labels: [ __meta_kubernetes_pod_name ]
+    target_label: pod
+  metric_relabel_configs:
+  - source_labels: [ __name__ ]
+    action: keep
+    regex: ^(` + monitoringMetricSchedulerBindingLatencyMicrosecondsBucket + `|` + monitoringMetricSchedulerE2eSchedulingLatencyMicrosecondsBucket + `|` + monitoringMetricSchedulerSchedulingAlgorithmLatencyMicrosecondsBucket + `|` + monitoringMetricRestClientRequestsTotal + `|` + monitoringMetricProcessMaxFds + `|` + monitoringMetricProcessOpenFds + `)$`
```

Comment:
wouldn't it be better to have a slice of whitelisted metrics and then join them with `|`?
```suggestion
    regex: ^(` + strings.Join(whitelistedMetrics, "|") + `)$`
```

Reply:
Yes, makes sense. I moved this out from the `const` block to the `var` block and implemented your suggestion.